### PR TITLE
feat: Mark V1 ready preview resources

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- mark preview resources
- add a separate flag for enabling all of the resources to simplify tests

## Test Plan

* [ ] acceptance tests

* [ ] …

## References

*

### CODE REVIEW

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/datasources/current_account.go [hunk 1]
```diff
@@ -41,6 +42,10 @@ func CurrentAccount() *schema.Resource {
 // ReadCurrentAccount read the current snowflake account information.
 func ReadCurrentAccount(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*provider.Context).Client
+	enabled := meta.(*provider.Context).EnabledFeatures
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/datasources/current_account.go:45]: If we extract them as this fancy closed list then we need conversion from string to this new type, preferably inside this new pkg, so this validation in provider changes, sorry for not being precise

File: pkg/provider/provider.go [hunk 2]
```diff
@@ -364,6 +373,59 @@ func Provider() *schema.Provider {
 	}
 }
 
+var previewFeatures = []string{
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/provider.go:376]: That's interesting - they are preview data sources :) I'll change the values here to have a better separation of resources and data sources.

File: pkg/provider/provider.go [hunk 3]
```diff
@@ -535,20 +597,42 @@ func ConfigureProvider(ctx context.Context, s *schema.ResourceData) (any, diag.D
 
 	client, clientErr := sdk.NewClient(config)
 
+	providerCtx := &provider.Context{Client: client}
+
+	if v, ok := s.GetOk("preview_features_enabled"); ok {
+		providerCtx.EnabledFeatures = expandStringList(v.(*schema.Set).List())
+	}
+
+	if os.Getenv("SF_TF_ACC_TEST_ENABLE_ALL_PREVIEW_FEATURES") == "true" {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/provider.go:606]: I'll leave it as is, to have similar behavior to `SF_TF_ACC_TEST_CONFIGURE_CLIENT_ONCE`.

[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/provider.go]: Done.

File: pkg/provider/provider.go [hunk 4]
```diff
@@ -356,6 +356,15 @@ func Provider() *schema.Provider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc(snowflakeenvs.Profile, "default"),
 			},
+			"preview_features_enabled": {
+				Type:     schema.TypeList,
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/provider.go]: Done.

File: pkg/provider/provider.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/provider.go]: Done.

**Review by sfc-gh-asawicki — APPROVED** [review]

File: pkg/internal/tracking/context.go [hunk 5]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 const (
-	ProviderVersion string = "v0.99.0" // TODO(SNOW-1814934): Currently hardcoded, make it computed
+	ProviderVersion string = "v1.0.0-rc.1" // TODO(SNOW-1814934): Currently hardcoded, make it computed
```
Comment (sfc-gh-jcieslak) [inline pkg/internal/tracking/context.go:11]: make it `v1.0.0-rc1`

File: MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -9,6 +9,67 @@ across different versions.
...
+	- `snowflake_shares`
+	- `snowflake_parameters`
+	- `snowflake_stages`
+	- `snowflake_storage_integrations`
+	- `snowflake_system_generate_scim_access_token`
+	- `snowflake_system_get_aws_sns_iam_policy`
+	- `snowflake_system_get_privatelink_config`
+	- `snowflake_system_get_snowflake_platform_info`
+
+If you want to have them enabled, add the feature name to the provider configuration (with `_datasource` or `_resource` suffix), like this:
```
Comment (sfc-gh-jcieslak) [inline MIGRATION_GUIDE.md:66]: Mention the option to enable all features at once.

File: pkg/internal/tracking/context.go (hunk 5 above)
Comment (sfc-gh-jmichalak) [reply pkg/internal/tracking/context.go:11]: I initially did, but realized that in semantic versioning they use `.` here: https://semver.org/

File: MIGRATION_GUIDE.md (hunk 6 above)
Comment (sfc-gh-jmichalak) [reply MIGRATION_GUIDE.md:66]: It's not available for the customers, and done only to simplify the provider configuration setup.

File: pkg/internal/tracking/context.go (hunk 5 above)
Comment (sfc-gh-asawicki) [reply pkg/internal/tracking/context.go:11]: `A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version.` (https://semver.org/#spec-item-9) - so in our case I would say that rc1 is and identifier, so I agree with @sfc-gh-jcieslak

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/internal/tracking/context.go (hunk 5 above)
Comment (sfc-gh-jmichalak) [reply pkg/internal/tracking/context.go:11]: I'll change this in the next pr.

**Pull request merged**